### PR TITLE
drainer: support concurrent DDL

### LINKED ISSUE #1179: drainer is not compatibile with (concurrent DDL/ v6.2.0) now
## Bug Report

Please answer these questions before submitting your issue. Thanks!

1. What did you do?
If possible, provide a recipe for reproducing the error.
nightly-with-concurrent-ddl -> v5.4.0
drainer can't get start.
image

2. What did you expect to see?

3. What did you see instead?

4. Please provide the relate downstream type and version of drainer.
(run `drainer -V` in terminal to get drainer's version)

### PR DESCRIPTION
**Pull request by wjhuang2016**
Signed-off-by: wjhuang2016 <huangwenjun1997@gmail.com>

### What problem does this PR solve?

Issue Number: close #1179

### What is changed and how it works?
Init `mysql` schema after creating a new schema.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)
According to #1179, the drainer can get started and work after this PR.
Side effects

Related changes

### Release note

- No release note

### CODE REVIEW

File: drainer/schema.go [hunk 1]
```diff
@@ -78,6 +78,19 @@ func NewSchema(jobs []*model.Job, hasImplicitCol bool) (*Schema, error) {
...
+// InitForCreateMySQLSchema create the schema info for `mysql`, since it's created by KV after TiDB 6.2.
+func (s *Schema) InitForCreateMySQLSchema() {
+	db := model.DBInfo{
+		ID:      1,
+		Name:    model.NewCIStr(mysql.SystemDB),
+		Charset: mysql.UTF8MB4Charset,
+		Collate: mysql.UTF8MB4DefaultCollation,
+		State:   model.StatePublic,
+	}
+	s.schemas[1] = &db
```
Comment (xiongjiwei) [inline drainer/schema.go:90]: schema id may not be 1 if it is an upgraded cluster

File: drainer/schema.go [hunk 2]
```diff
@@ -78,6 +78,19 @@ func NewSchema(jobs []*model.Job, hasImplicitCol bool) (*Schema, error) {
...
 }
 
+// InitForCreateMySQLSchema create the schema info for `mysql`, since it's created by KV after TiDB 6.2.
+func (s *Schema) InitForCreateMySQLSchema() {
+	db := model.DBInfo{
+		ID:      1,
+		Name:    model.NewCIStr(mysql.SystemDB),
+		Charset: mysql.UTF8MB4Charset,
+		Collate: mysql.UTF8MB4DefaultCollation,
+		State:   model.StatePublic,
```
Comment (lichunzhu) [inline drainer/schema.go:88]: Can we use a tidb global variable to save this?

File: drainer/schema.go [hunk 3]
```diff
@@ -78,6 +78,19 @@ func NewSchema(jobs []*model.Job, hasImplicitCol bool) (*Schema, error) {
...
+	}
+	s.schemas[1] = &db
+	s.schemaNameToID["mysql"] = 1
```
Comment (lichunzhu) [inline drainer/schema.go:91]:
```suggestion
	s.schemaNameToID[mysql.SystemDB] = 1
```

File: drainer/schema.go [hunk 4]
```diff
@@ -156,8 +169,14 @@ func (s *Schema) DropSchema(id int64) (string, error) {
 
 // CreateSchema adds new DBInfo
 func (s *Schema) CreateSchema(db *model.DBInfo) error {
-	if _, ok := s.schemas[db.ID]; ok {
-		return errors.AlreadyExistsf("schema %s(%d)", db.Name, db.ID)
+	if dbInfo, ok := s.schemas[db.ID]; ok {
+		// The `mysql` database is created by SQL, not by KV.
+		if dbInfo.Name.L == "mysql" {
```
Comment (lichunzhu) [inline drainer/schema.go:174]: ditto

File: drainer/schema.go (hunk 1 above)
Comment (lichunzhu) [reply drainer/schema.go:90]: How about checking `jobs` and then decide whether to add this job or not?

Comment (wjhuang2016) [reply drainer/schema.go:90]: It's ok for now since it's covered by new schema if it is an upgraded cluster

File: drainer/schema.go [hunk 5]
```diff
@@ -78,6 +87,12 @@ func NewSchema(jobs []*model.Job, hasImplicitCol bool) (*Schema, error) {
 	return s, nil
 }
 
+// InitForCreateMySQLSchema create the schema info for `mysql`, since it's created by KV after TiDB 6.2.
+func (s *Schema) InitForCreateMySQLSchema() {
+	s.schemas[1] = &defaultMySQLSchemaInfo
+	s.schemaNameToID[mysql.SystemDB] = 1
```
Comment (lichunzhu) [inline drainer/schema.go:93]:
```suggestion
	s.schemas[defaultMySQLSchemaInfo.ID] = &defaultMySQLSchemaInfo
	s.schemaNameToID[mysql.SystemDB] = defaultMySQLSchemaInfo.ID
```

**Review by lichunzhu — APPROVED** [review]
Rest LGTM

File: drainer/schema.go [hunk 6]
```diff
@@ -28,6 +28,15 @@ import (
 const implicitColName = "_tidb_rowid"
 const implicitColID = -1
 
+// Keep it's the same as TiDB side.
+var defaultMySQLSchemaInfo = model.DBInfo{
```
[the remark this answers is not in the document]
Comment (wjhuang2016) [reply drainer/schema.go:32]: No, we can export it later.

**Review by glorv — APPROVED** [review]

**Pull request merged**